extensions/ql-vscode/test/unit-tests/packages/commands/CommandManager.test.ts

```diff
@@ -1,8 +1,111 @@
-import { CommandManager } from "../../../../src/packages/commands";
+import {
+  CommandFunction,
+  CommandManager,
+} from "../../../../src/packages/commands";
 
-describe(CommandManager.name, () => {
-  it("can create a command manager", () => {
-    const commandManager = new CommandManager();
-    expect(commandManager).not.toBeUndefined();
+describe("CommandManager", () => {
+  it("can register a command", () => {
+    const commandRegister = jest.fn();
+    const commandManager = new CommandManager<Record<string, CommandFunction>>(
+      commandRegister,
+      jest.fn(),
+    );
+    const myCommand = jest.fn();
+    commandManager.registerCommand("abc", myCommand);
+    expect(commandRegister).toHaveBeenCalledTimes(1);
+    expect(commandRegister).toHaveBeenCalledWith("abc", myCommand);
+  });
+
+  it("can register typed commands", async () => {
+    const commands = {
+      "codeQL.openVariantAnalysisLogs": async (variantAnalysisId: number) => {
+        return variantAnalysisId * 10;
+      },
+    };
+    const commandManager = new CommandManager<typeof commands>(
+      jest.fn(),
+      jest.fn(),
+    );
+
+    // @ts-expect-error wrong command name should give a type error
+    commandManager.registerCommand("abc", jest.fn());
```

Comment:
It is a good question. I wasnt sure, but given how simple the test is I thought it actually a quite neat way of making sure it actually behaves as expected. E.g. if we accidentally had an `any` somewhere in the implementation, then it might suddenly not give us the safety we expect.

Reply:
Yeah I guess if the point is for us to assure ourselves that our typing works as we expect it to, then it makes sense. But I would argue if we had to add tests on top of the type system, then why do we need the type system in the first place. It seems like tests are doing the validation for us. 

But this is getting philosophical, I still think what you're doing makes sense. It's just annoying that we can't trust it completely.

Reply:
I don't think this is really about testing the type system. We're basically testing that the types that we have defined are working and correctly give errors where we expect them to. This basically ensure that if we change the first parameter of `registerCommand` to be `string` instead of `CommandName`, we get this error. I don't think there's another way to test this, but we would like to have that assurance.

Reply:
Yeah I understand that we want to see that it actually works as expected, which is why I didn't press it further. 

But I'll be honest, it still a bit weird to me to have to test Typescript. We're intentionally disabling typescript errors to make the test work. I understand we could miss something if we redefined the types, but we don't really do this kind of testing pattern for other Typescript constraints, and those could also start failing at any time. 

I would be inclined to trust the type system, the way we do for the rest of the codebase. But I understand if you want extra assurance.